src/krbd.cc

```diff
@@ -272,6 +273,12 @@ static int build_map_buf(CephContext *cct, const krbd_spec& spec,
 
   if (!options.empty())
     oss << "," << options;
+
+  /* setting default messanger: msgr2 in prefer-crc mode */
+  if (msgr2 && !ms_mode_specified) {
```

Comment:
Would checking just `ms_mode_specified` should be sufficient here?  I don't see a way for `msgr2` to be false when `ms_mode_specified` is also false:
```suggestion
  if (!ms_mode_specified) {
```